src/router.js

```diff
@@ -69,7 +99,7 @@ const routes = [
     path: '/lobby/:gameId',
     component: LobbyView,
     // TODO: Add logic to redirect if a given game does not exist
-    beforeEnter: mustBeAuthenticated,
+    beforeEnter: checkAndSubscribeToLobby,
```

Comment:
This means that we double request to join the game, right? Once from the home page before the router push and then right here before the page is loaded by the router. I'm wondering how we can nicely hit it once instead. @seriouslysean penny for your thoughts:

I think we want the home page to be able to show a loading state when you hit the join button and keep you on the home page until the you successfully subscribe to the game so that if the subscribe call fails, you can stay on the home page and get an error snackbar without being redirected to an error view and then having to come back.

But if you navigate directly to the lobby url, how do we subscribe you to the game in a way that isn't redundant to the flow where you're clicking the join button from the home page?

Reply:
I think the move may be to set a piece of store state when coming from the home page that this nav guard checks from and uses to skip the subscription call in that case

Reply:
I don't believe it double requests, After the initial request it triggers `gameStore.updateGame()` which should update the players in gameStore, which this function checks against the username, so if the user is already apart of the game we give the go ahead, if not we request subscribe

Reply:
I talked to @itsalaidbacklife about this a little bit. It feels like all (or most) of the logic should be occurring here so it fires for both direct loads, and client side loads.

In my opinion, the store should handle the state of the incoming user and whether here or in the component you’ll be able to map the data you need without duplication.

This will also allow you to utilize a bit of state for the toast message if the game can’t be found or it otherwise invalid.